### PR TITLE
LPC55xxx secure/non-secure split

### LINKED ISSUE #16330: LPCXpresso55S69 secure/non-secure configuration
we need to look at structuring this similar to how dts/arm/nordic/nrf9160* is done.  So that secure/non-secure can work cleanly in the future.

_Originally posted by @galak in https://github.com/zephyrproject-rtos/zephyr/pull/15647_

Memory and peripherals need to be configured as both secure and non-secure.

### PR DESCRIPTION
**Pull request by agansari**
Adds NS definitions for CPU0 and CPU1.
Preparing platform for future development, dual core, mpu, etc.

Fixes #16330

### CODE REVIEW

File: boards/arm/lpcxpresso55s69/lpcxpresso55s69_cpu0.dts [hunk 1]
```diff
@@ -64,14 +65,49 @@
...
+			label = "image-0";
+			reg = <0x10020000 0x00020000>;
+		};
+		slot1_partition: partition@10040000 {
+			label = "image-1";
+			reg = <0x10040000 0x00020000>;
+		};
+		scratch_partition: partition@10060000 {
+			label = "image-scratch";
+			reg = <0x10060000 0x0003D800>;
```
Comment (d3zd3z) [inline boards/arm/lpcxpresso55s69/lpcxpresso55s69_cpu0.dts]: I believe the DT convention is to use lowercase for hex constants.

File: dts/arm/nxp/nxp_lpc55S6x_ns.dtsi [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+			flash0:flash@0{
+				compatible = "soc-nv-flash";
+				label = "MCUX_FLASH";
+				reg = <0x0 DT_SIZE_K(630)>;
+				erase-block-size = <512>;
+				write-block-size = <512>;
+			};
+
+			flash_reserved:flash@9D800 {
```
Comment (d3zd3z) [inline dts/arm/nxp/nxp_lpc55S6x_ns.dtsi]: This hex value, and the one below should be lowercase.

File: dts/arm/nxp/nxp_lpc55S6x_ns.dtsi [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+			boot_rom:flash@3000000 {
+				compatible = "soc-nv-flash";
+				reg = <0x3000000 DT_SIZE_K(128)>;
+			};
+		};
+
+		usart0:usart@40086000 {
+			compatible = "nxp,lpc-usart";
+			reg = <0x40086000 0xE44>;
```
Comment (d3zd3z) [inline dts/arm/nxp/nxp_lpc55S6x_ns.dtsi]: Likewise. Lowercase.

**Review by d3zd3z — CHANGES_REQUESTED** [review]
Other than a minor issue with the case of hex values, this looks good.

File: boards/arm/lpcxpresso55s69/lpcxpresso55s69_cpu1.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#
+# Copyright (c) 2019, NXP
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+identifier: lpcxpresso55s69_cpu0
```
Comment (microbuilder) [inline boards/arm/lpcxpresso55s69/lpcxpresso55s69_cpu1.yaml]: I may have misunderstood something, but should this be `lpcxpresso55s69_cpu1` if the NS app is running on the second core?

File: boards/arm/lpcxpresso55s69/lpcxpresso55s69_ns.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+#
+# Copyright (c) 2019, NXP
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+identifier: lpcxpresso55s69_cpu0
```
Comment (microbuilder) [inline boards/arm/lpcxpresso55s69/lpcxpresso55s69_ns.yaml]: As above, should this be `lpcxpresso55s69_cpu1` if the NS app is running on the second core?

Comment (agansari) [reply boards/arm/lpcxpresso55s69/lpcxpresso55s69_ns.yaml]: @microbuilder lpcxpresso55s69_ns refers to CPU0 built with NS address ranges.
CPU1 is NS only, guess I should document this.

**Comment by agansari** [comment]
@microbuilder added documentation about the possible targets

**Review by microbuilder — APPROVED** [review]

File: dts/arm/nxp/nxp_lpc55S6x_ns.dtsi [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+	soc {
+		flash-controller@40034000 {
+			compatible = "nxp,lpc-iap";
+			label = "FLASH_IAP";
+			reg = <0x40034000 0x18>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			flash0:flash@0{
```
Comment (d3zd3z) [inline dts/arm/nxp/nxp_lpc55S6x_ns.dtsi]: There should be a space after the colon (here and several below).

**Review by d3zd3z — CHANGES_REQUESTED** [review]
One minor whitespace issue, but otherwise looks good.

**Review by d3zd3z — APPROVED** [review]

**Pull request merged**